### ISSUE TITLE
Handlers receive empty dict for status when status is not set

### ISSUE THREAD

**Issue by jkupferer** [zalando-incubator/kopf#183]
## Expected Behavior

If a resource has no status then any handler should receive `None` for `status` and the `body` argument to the handler should have no key `status`.

## Actual Behavior

Handlers that are processing resources with no status receive `status` as an empty dictionary and the `body` has an empty dictionary for `status`.

This causes problems when calculating how to patch a resource to set a value in the status.

For example, to set `{"foo": "bar"}` in the status of a resource with no status subresource requires jsonpatch:

```
[{"op":"add", "path":"/status", "value": {"foo":"bar"}}]
```

However if the status subresource exists and is empty, the jsonpatch required is:

```
[{"op":"add", "path":"/status/foo", "value": "bar"}]
```

The current handling of status prevents detecting and providing the correct patch.

**Comment by nolar** [comment]
Presumably solved with #198 — now, for `spec`, `status`, and `meta` kwargs, a lazy dict resolver is passed — which renders, reprs, and behaves as these stenzas were dicts, but without actually adding them to the `body` object via `body.setdefault()`. Therefore, those who use the `body` kwarg for the objects without these stenzas, will not see them added implicitly.

Tested locally on a sample operator — works fine, debugging is also fine, all is good. But some side-effects are possible — to be tested in action in the bigger operators (only those that use these 3 magic kwargs).

**Comment by nolar** [comment]
Pre-released as `kopf>=0.23rc1`. 

I assume the issue is fixed. If not, feel free to re-open the issue or at least to add a comment.

And thank you for your feedback! 😄
